### PR TITLE
Add Management API Anomaly endpoints

### PR DESCRIPTION
**Pull request by makoto-matsumoto**
### Changes

Add endpoint for Anomaly API

### References

https://auth0.com/docs/api/management/v2#!/Anomaly/get_ips_by_id
https://auth0.com/docs/api/management/v2#!/Anomaly/delete_ips_by_id

### Testing

* [x] This change adds unit test coverage
* [x] This change adds integration test coverage
* [x] This change has been tested on the latest version of Ruby

### Checklist

* [x] I have read the [Auth0 general contribution guidelines](https://github.com/auth0/open-source-template/blob/master/GENERAL-CONTRIBUTING.md)
* [x] I have read the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md)
* [x] All existing and new tests complete without errors
* [x] Rubocop passes on all added/modified files
* [x] All active GitHub checks have passed

### CODE REVIEW

File: spec/integration/lib/auth0/api/v2/api_anomaly_spec.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
+require 'spec_helper'
+describe Auth0::Api::V2::Anomaly do
+  let(:client) { Auth0Client.new(v2_creds) }
+  let(:ip) { '192.0.2.0' }
+
+  describe '.check_if_ip_is_blocked', vcr: true do
+    it 'should rerurn 200 response code' do
```
Comment (joshcanhelp) [inline spec/integration/lib/auth0/api/v2/api_anomaly_spec.rb]:
```suggestion
    it 'should return 200 response code' do
```

File: lib/auth0/api/v2/anomaly.rb [hunk 2]
```diff
@@ -0,0 +1,36 @@
+module Auth0
+  module Api
+    module V2
+      # Methods to use the anomaly endpoints
+      module Anomaly
+        # Use this route to determine if a given ip is currently blocked
```
Comment (joshcanhelp) [inline lib/auth0/api/v2/anomaly.rb]:
```suggestion
        # Use this route to determine if a given IP is currently blocked
```

File: spec/fixtures/vcr_cassettes/Auth0_Api_V2_Anomaly/_check_if_ip_is_blocked/should_rerurn_200_response_code.yml [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+      Accept:
+      - "*/*"
+      Accept-Encoding:
+      - gzip, deflate
+      User-Agent:
+      - rest-client/2.0.2 (darwin18.6.0 x86_64) ruby/2.6.3p62
+      Content-Type:
+      - application/json
+      Auth0-Client:
+      - eyJuYW1lIjoicnVieS1hdXRoMCIsInZlcnNpb24iOiI0LjUuMCJ9
```
Comment (joshcanhelp) [inline spec/fixtures/vcr_cassettes/Auth0_Api_V2_Anomaly/_check_if_ip_is_blocked/should_rerurn_200_response_code.yml]: Your attention to detail is commendable!

I'll make a note to anonymize this in our existing cassettes. Nothing secret here but probably best not to have specific application data in here.

File: lib/auth0/api/v2/anomaly.rb [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+        # by the failed login to multiple user accounts trigger.
+        # @see https://auth0.com/docs/api/management/v2#!/Anomaly/get_ips_by_id
+        # @param ip [string] The IP to check.
+        def check_if_ip_is_blocked(ip)
+          raise Auth0::InvalidParameter, 'Must specify an IP' if ip.to_s.empty?
```
[the remark this answers is not in the document]
Comment (makoto-matsumoto) [reply lib/auth0/api/v2/anomaly.rb]: I think that it is better to return the original response of Auth0 as much as possible.
If the format is incorrect, Auth0 will return:

```
RESPONSE CODE : 400
RESPONSE BODY
{
  "statusCode": 400,
  "error": "Bad Request",
  "message": "Path validation error: 'Object didn't pass validation for format ipv4: 9919191' on property id (The ip to check).",
  "errorCode": "invalid_uri"
}
```

File: spec/fixtures/vcr_cassettes/Auth0_Api_V2_Anomaly/_check_if_ip_is_blocked/should_rerurn_200_response_code.yml (hunk 3 above)
Comment (makoto-matsumoto) [reply spec/fixtures/vcr_cassettes/Auth0_Api_V2_Anomaly/_check_if_ip_is_blocked/should_rerurn_200_response_code.yml]: Thank you for your advice!  
However, this value has been updated to the same value as the test already included in the master.
Ex:
[should_add_a_token_to_the_blacklist.yml:19](https://github.com/auth0/ruby-auth0/blob/master/spec/fixtures/vcr_cassettes/Auth0_Api_V2_Blacklists/_add_token_to_blacklist/should_add_a_token_to_the_blacklist.yml#L19)

You might want to include client id in filter_sensitive_data as well.

spec/spec_helper.rb
``` 
config.filter_sensitive_data('CLIENT_SECRET') { ENV['CLIENT_SECRET'] }
config.filter_sensitive_data('API_TOKEN') { ENV['MASTER_JWT'] }
```

Comment (joshcanhelp) [reply spec/fixtures/vcr_cassettes/Auth0_Api_V2_Anomaly/_check_if_ip_is_blocked/should_rerurn_200_response_code.yml]: I think that's probably the best way to deal with it. My main reason for commenting was that I appreciate you updating to the same value as the rest 👍

File: lib/auth0/api/v2/anomaly.rb (hunk 4 above)
[the remark this answers is not in the document]
Comment (joshcanhelp) [reply lib/auth0/api/v2/anomaly.rb]: The main thing I was thinking here is that we could save an HTTP call if we can catch the error (like we do in other cases). I'll check with the team to see what they think.

**Pull request merged**